### PR TITLE
Stub receives LocalSlamResults.

### PR DESCRIPTION
**Pull request by gaschler**
[RFC=0002](https://github.com/googlecartographer/rfcs/blob/master/text/0002-cloud-based-mapping-1.md)
PAIR=cschuet

### CODE REVIEW

File: cartographer_grpc/mapping/trajectory_builder_stub.cc [hunk 1]
```diff
@@ -23,13 +23,48 @@ namespace cartographer_grpc {
...
+    proto::ReceiveLocalSlamResultsRequest request;
+    request.set_trajectory_id(trajectory_id);
+    local_slam_result_reader_.client_reader = stub_->ReceiveLocalSlamResults(
+        &local_slam_result_reader_.client_context, request);
+    auto* client_reader_ptr = local_slam_result_reader_.client_reader.get();
+    local_slam_result_reader_.thread =
+        cartographer::common::make_unique<std::thread>(
+            [client_reader_ptr, local_slam_result_callback]() {
+              proto::ReceiveLocalSlamResultsResponse response;
+              while (client_reader_ptr->Read(&response)) {
```
Comment (cschuet) [inline cartographer_grpc/mapping/trajectory_builder_stub.cc]: Too much indentation.

Comment (gaschler) [reply cartographer_grpc/mapping/trajectory_builder_stub.cc]: Done, moved to a static function.

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**